### PR TITLE
junit5: Parameterize the JUnit 5 artifact versions

### PR DESCRIPTION
**Pull request by bjhargrave**
We need different versions for our unit testing needs than we need for
building our JUnit 5 testing support.

This updates JUnit 5 to 5.6 for our unit tests.

### CODE REVIEW

**Comment by bjhargrave** [comment]
@kriegfrj It would be nice to avoid `junit.platform.tester.version=1.5.2` for biz.aQute.tester.test but using `1.6.0` causes NoClassDefFoundErrors when running the tests. The 1.6.0 launcher expects a type from the 1.6.0 engine and the test comment indicate we must use the same engine version as Eclipse.

File: cnf/ext/central.mvn [hunk 1]
```diff
@@ -62,14 +62,18 @@ junit:junit:${junit4.tester.version}
...
+org.junit.platform:junit-platform-launcher:${junit.platform.tester.version}
+org.junit.platform:junit-platform-testkit:${junit.platform.tester.version}
+org.junit.jupiter:junit-jupiter-api:${junit.jupiter.version}
+org.junit.jupiter:junit-jupiter-engine:${junit.jupiter.version}
+org.junit.jupiter:junit-jupiter-params:${junit.jupiter.version}
+org.junit.vintage:junit-vintage-engine:${junit.jupiter.version}
+org.junit.platform:junit-platform-commons:${junit.platform.version}
+org.junit.platform:junit-platform-engine:${junit.platform.version}
+org.junit.platform:junit-platform-launcher:${junit.platform.version}
+org.junit.platform:junit-platform-testkit:${junit.platform.version}
```
[the remark this answers is not in the document]
Comment (bjhargrave) [reply cnf/ext/central.mvn]: Fixed.

**Comment by kriegfrj** [comment]
I had a bit of a play, but I don't think there's a lot we can do about running `junit.platform.tester.version` on 1.5.2. 

Btw, the comment that says the tester version must be the same as the Eclipse version was because I reasoned that, because of the runtime dependencies that are brought in during testing, it would be better to test against the lowest version at runtime in case we end up accidentally depending on more advanced features. However, #3745 has shown that that approach also has weaknesses. Ideally, we'd test against the highest *and* the lowest (and maybe a couple of versions in-between). This is not easily possible when using the sharing feature of Launchpad because you have to have the same version on the `-testpath` as at runtime. However, it might be possible by refactoring some of the tests and using `addResourceWithCopy()` and other tricks - using parameterized tests, you could make the JUnit version a parameter and pass that in to the LaunchpadBuilder so that it installs a particular version.

Anyway, I think that these are possible enhancements for another day and that what you now have is the best we can do without making wholesale changes.

**Comment by bjhargrave** [comment]
> I think that these are possible enhancements for another day

Thanks for thinking about this.

**Pull request merged**